app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt

```diff
@@ -0,0 +1,201 @@
+    private var userName: String = ""
+    private var topic: TalkPage.Topic? = null
+    private var replyActive = false
+    private val bottomSheetPresenter = ExclusiveBottomSheetPresenter()
+
+    private var linkHandler: TalkLinkHandler? = null
+    private val linkMovementMethod = LinkMovementMethodExt { url: String ->
+        linkHandler?.onUrlClick(url, null, "")
+    }
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_talk_topic)
+        title = ""
+        linkHandler = TalkLinkHandler(this)
+
+        if (intent.hasExtra(EXTRA_LANGUAGE)) {
+            wikiSite = WikiSite.forLanguageCode(intent.getStringExtra(EXTRA_LANGUAGE).orEmpty())
+        }
+        userName = intent.getStringExtra(EXTRA_USER_NAME).orEmpty()
+        topicId = intent.extras?.getInt(EXTRA_TOPIC, 0)!!
+
+        talkRecyclerView.layoutManager = LinearLayoutManager(this)
+        talkRecyclerView.addItemDecoration(DrawableItemDecoration(this, R.attr.list_separator_drawable, drawStart = false, drawEnd = false))
+        talkRecyclerView.adapter = TalkReplyItemAdapter()
+
+        L10nUtil.setConditionalLayoutDirection(talkRefreshView, wikiSite.languageCode())
+
+        talkReplyButton.setOnClickListener {
+            // TODO
+        }
+
+        talkRefreshView.setOnRefreshListener {
+            loadTopic()
+        }
+
+        talkReplyButton.visibility = View.GONE
+        loadTopic()
+    }
+
+    public override fun onDestroy() {
+        disposables.clear()
+        super.onDestroy()
+    }
+
+    private fun loadTopic() {
+        disposables.clear()
+        talkProgressBar.visibility = View.VISIBLE
+        talkErrorView.visibility = View.GONE
+
+        disposables.add(ServiceFactory.getRest(wikiSite).getTalkPage(userName)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ response ->
+                    topic = response.topics?.find { t -> t.id == topicId }
+                    updateOnSuccess()
+                }, { t ->
+                    L.e(t)
+                    updateOnError(t)
+                }))
+    }
+
+    private fun updateOnSuccess() {
+        talkProgressBar.visibility = View.GONE
+        talkErrorView.visibility = View.GONE
+        talkReplyButton.show()
+        talkRefreshView.isRefreshing = false
+
+        val titleStr = StringUtil.fromHtml(topic?.html).toString().trim()
+        title = if (titleStr.isNotEmpty()) titleStr else getString(R.string.talk_no_subject)
+        talkRecyclerView.adapter?.notifyDataSetChanged()
+    }
+
+    private fun updateOnError(t: Throwable) {
+        talkProgressBar.visibility = View.GONE
+        talkRefreshView.isRefreshing = false
+        talkReplyButton.hide()
+        talkErrorView.visibility = View.VISIBLE
+        talkErrorView.setError(t)
```

Comment:
Should we set up the `setBackClickListener` and `setRetryClickListener` for the `talkErrorView`?

Reply:
ok!